src/pkg/packager/remove.go

```diff
@@ -36,71 +36,89 @@ func (p *Packager) Remove(packageName string) error {
-
-			if slices.Contains(requestedComponents, installedComponent.Name) {
-				for _, installedChart := range installedComponent.InstalledCharts {
-					helmCfg := helm.Helm{}
-					helmCfg.RemoveChart(installedChart.Namespace, installedChart.ChartName, spinner)
-				}
-
-				// Remove the component we just removed from the array
-				packages.DeployedComponents = append(packages.DeployedComponents[:i], packages.DeployedComponents[i+1:]...)
-			}
+	partialRemove := len(requestedComponents) > 0 && requestedComponents[0] != ""
 
-			if len(packages.DeployedComponents) == 0 {
-				// All the installed components were deleted, there for this package is no longer actually deployed
-				_ = p.cluster.Kube.DeleteSecret(packageSecret)
-			} else {
-				// Save the new secret with the removed components removed from the secret
-				newPackageSecret := p.cluster.Kube.GenerateSecret("zarf", secretName, corev1.SecretTypeOpaque)
-				newPackageSecret.Labels["package-deploy-info"] = p.cfg.Pkg.Metadata.Name
-				newPackageSecretData, _ := json.Marshal(packages)
-				newPackageSecret.Data["data"] = newPackageSecretData
-				err = p.cluster.Kube.ReplaceSecret(newPackageSecret)
-				if err != nil {
-					message.Warnf("Unable to replace the %s package secret: %#v", secretName, err)
-				}
-			}
+	for _, c := range utils.Reverse(packages.DeployedComponents) {
+		// Only remove the component if it was requested or if we are removing the whole package
+		if partialRemove && !slices.Contains(requestedComponents, c.Name) {
+			continue
+		}
+
+		if err := p.removeComponent(packages.Data.Components, c, spinner); err != nil {
+			return fmt.Errorf("unable to remove the component (%s): %w", c.Name, err)
 		}
-	} else {
-		// Loop through all the installed components and remove them
-		for i := len(packages.DeployedComponents) - 1; i >= 0; i-- {
-			installedComponent := packages.DeployedComponents[i]
-
-			// This component was installed onto the cluster. Prompt the user to see if they would like to remove it!
-			for _, installedChart := range installedComponent.InstalledCharts {
-				spinner.Updatef("Uninstalling chart (%s) from the (%s) component", installedChart.ChartName, installedComponent.Name)
-
-				helmCfg := helm.Helm{}
-				err = helmCfg.RemoveChart(installedChart.Namespace, installedChart.ChartName, spinner)
-				if err != nil {
-					message.Errorf(err, "Unable to remove the installed helm chart (%s) from the namespace (%s) of component (%s) (were dependent components removed first?)",
-						installedChart.ChartName, installedChart.Namespace, installedComponent.Name)
-
-					return err
-				}
+
+		// Remove the component we just removed from the array
+		packages.DeployedComponents = utils.RemoveMatches(packages.DeployedComponents, func(t types.DeployedComponent) bool {
+			return t.Name == c.Name
+		})
+
+		if len(packages.DeployedComponents) < 1 {
+			// All the installed components were deleted, there for this package is no longer actually deployed
```

Comment:
```suggestion
			// All the installed components were deleted, therefore this package is no longer actually deployed
```

Reply:
```suggestion
			// All the installed components were deleted, therefore this package is no longer actually deployed
```
Not sure why this was marked resolved.  `there for` is still in the comment.